spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/security/CognitoAuthenticationProperties.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.autoconfigure.security;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * Cognito Authentication properties.
+ *
+ * @author Eddú Meléndez
+ * @since 2.3
+ */
+@ConfigurationProperties(prefix = "spring.cloud.aws.security.cognito")
+public class CognitoAuthenticationProperties {
+
+	private static final String COGNITO_ISSUER = "https://cognito-idp.%s.amazonaws.com/%s";
```

Comment:
There should be a way to use it with Localstack, ideally that from the user point of view same client configuration principles apply (setting endpoint through `spring.cloud.aws.endpoint` etc).

Reply:
`issuerUrl` and `registryUrl` has been added